src/ActivityRepo.js

```diff
@@ -0,0 +1,56 @@
+class ActivityRepo {
+  constructor(activies) {
+    this.allActivities = activies;
+  }
+  gatherMilesWalked(user, date) {
+    let specificDay = this.allActivities.find(day => day.date === date && day.userID === user.id);
+    let int = ((user.strideLength * specificDay.numSteps) / 5280).toFixed(1);
+    return parseFloat(int);
+  }
+  gatherMinutesActive(user, date) {
+    let specificDay = this.allActivities.find(day => day.date === date && day.userID === user.id);
+    return specificDay.minutesActive;
+  }
+  calcAverageMinActiveForAWeek(user, date) {
+    let userActivities = this.allActivities.filter(activity => activity.userID === user.id);
+    let desiredDateIndex = userActivities.map(activity => activity.date).indexOf(date);
+    let desiredWeek = userActivities.slice(desiredDateIndex - 6, desiredDateIndex + 1);
+    let totalTimeActive = desiredWeek.reduce((totalMin, activity) => {
+      totalMin += activity.minutesActive;
+      return totalMin;
+    }, 0);
+    return parseFloat((totalTimeActive / 7).toFixed(1));
+  }
```

Comment:
Really awesome job using all of these iterator methods - those are hard to pick up but they are well-executed!